include/eve/algo/ptr_iterator.hpp

```diff
@@ -34,20 +35,21 @@ namespace eve::algo
       return aligned_ptr_iterator<T, cardinal>{eve::previous_aligned_address(ptr, cardinal{})};
     }
 
+    value_type& operator*() const { return *ptr; }
+
     unaligned_ptr_iterator& operator+=(std::ptrdiff_t n) { ptr += n; return *this; }
     friend std::ptrdiff_t   operator-(unaligned_ptr_iterator x, unaligned_ptr_iterator y) { return x.ptr - y.ptr; }
 
     auto operator<=>(unaligned_ptr_iterator const&) const = default;
 
-    template <relative_conditional_expr C>
-    friend wide_value_type tagged_dispatch( eve::tag::load_, C cond, unaligned_ptr_iterator self )
-    {
-      return eve::load[cond](self.ptr, cardinal{});
-    }
 
-    friend wide_value_type tagged_dispatch( eve::tag::load_, unaligned_ptr_iterator self )
+    template< relative_conditional_expr C, decorator S, typename Pack>
+    friend auto tagged_dispatch ( eve::tag::load_, C const& c, S const& s
+                                , eve::as_<Pack> const& , unaligned_ptr_iterator self
+                                )
     {
-      return eve::load(self.ptr, cardinal{});
+      using tgt_t = typename Pack::template rescale<Cardinal>;
```

Comment:
Hmm. I'm not sold on this.
Ideally it should be:

If there is no `cardinal`, pass a cardinal.

Hmm. Hmm. Leave as is, I'd have to think about this.

Reply:
Fixed